test-it/karma/fixtures/javascriptinvoke/CloseReference.via

```diff
@@ -0,0 +1,111 @@
+define(DynamicReferencesVI dv(VirtualInstrument (
+    Locals: c(
+        ce(dv(.ErrorCluster (true 100 'error' )) errorIn)
+        e(c(e(.Boolean status) e(.Int32 code) e(.String source)) err)
+        e(c(e(.Boolean status) e(.Int32 code) e(.String source)) getObjectReferenceError1)
+        e(c(e(.Boolean status) e(.Int32 code) e(.String source)) getObjectReferenceError2)
+
+        e(c(e(.Boolean status) e(.Int32 code) e(.String source)) noErrorInAndValidReferenceError)
+        e(c(e(.Boolean status) e(.Int32 code) e(.String source)) errorInAndValidReferenceError)
+        e(c(e(.Boolean status) e(.Int32 code) e(.String source)) noErrorInAndInvalidReferenceError)
+        e(c(e(.Boolean status) e(.Int32 code) e(.String source)) errorInAndInvalidReferenceError)
+
+        e(.JavaScriptDynamicRefNum objectReference)
+        e(dv(.Boolean false) isNotAValidRefnum1)
+        e(dv(.Boolean false) isNotAValidRefnum2)
+        e(dv(.Boolean false) isNotAValidRefnum3)
+        e(dv(.Boolean false) isNotAValidRefnum4)
+        e(.Occurrence occurrence)
+    )
+    clump (
+        // no error in, valid reference, close reference => no error out, reference closed
+        JavaScriptInvoke(occurrence false getObjectReferenceError1 'NI_GetObjectFunction' objectReference)
+        CloseReference(objectReference noErrorInAndValidReferenceError)
+        IsNotANumPathRefnum(objectReference isNotAValidRefnum1)
+
+        // error in, valid reference, close reference => error out same as error in, reference closed
+        JavaScriptInvoke(occurrence false getObjectReferenceError2 'NI_GetObjectFunction' objectReference)
+        Copy(errorIn errorInAndValidReferenceError)
+        CloseReference(objectReference errorInAndValidReferenceError)
+        IsNotANumPathRefnum(objectReference isNotAValidRefnum2)
+
+        // no error in, invalid reference, close reference => invalid reference error
+        CloseReference(objectReference noErrorInAndInvalidReferenceError)
```

Comment:
Not that necessary here. If you want to do for completion either here or preferably in G tests, just add for JS Reference that is a constant (defined with "ce" here) for similar tests.

Reply:
True, but in this case I wanted to use a reference that was previously closed.